c2pa_c_ffi/src/c_api.rs

```diff
@@ -431,6 +431,19 @@ pub unsafe extern "C" fn c2pa_string_free(s: *mut c_char) {
     }
 }
 
+/// Frees a const string allocated by Rust.
+///
+/// # Safety
+/// * Reads from NULL-terminated C strings.
+/// * The string can only be freed once and is invalid after this call.
+#[no_mangle]
+pub unsafe extern "C" fn c2pa_const_string_free(s: *const c_char) {
+    if !s.is_null() {
+        // Cast is okay as strings allocated by Rust and then passed to C are mutable.
+        drop(CString::from_raw(s as *mut c_char));
+    }
+}
+
```

Comment:
Hm, it doesn't seem right to make another string free function. Maybe we should instead be returning a `*mut ptr` from `Reader::remote_url` and save `*const ptr` for pointers that don't need to be freed.